services/repository.py

```diff
@@ -51,19 +52,62 @@ def get_repo_provider_service(
     commit=None,
     installation_name_to_use: Optional[str] = GITHUB_APP_INSTALLATION_DEFAULT_NAME,
 ) -> torngit.base.TorngitBaseAdapter:
+    """Gets the TorngitBaseAdapter instance for this repository.
+
+    !raises: RepositoryWithoutValidBotError
+    """
     _timeouts = [
         get_config("setup", "http", "timeouts", "connect", default=30),
         get_config("setup", "http", "timeouts", "receive", default=60),
     ]
-    service = repository.owner.service
+    service: str = repository.owner.service
     installation_info = get_owner_installation_id(
         repository.owner,
         repository.using_integration,
         repository=repository,
         ignore_installation=False,
         installation_name=installation_name_to_use,
     )
-    token, token_owner = get_repo_appropriate_bot_token(repository, installation_info)
+    try:
+        token, token_owner = get_repo_appropriate_bot_token(
+            repository, installation_info
+        )
+    except RepositoryWithoutValidBotError:
+        # GitHub exclusive bit -- START
```

Comment:
It's not a problem if we don't check cause it would just raise the exception again, but yeah, makes sense to do that as save a few operations... and have a clearer distinction of logic in the code.